docs/x509.rst

```diff
@@ -196,6 +196,24 @@ X.509 Certificate Object
             >>> isinstance(cert.signature_hash_algorithm, hashes.SHA256)
             True
 
+    .. attribute:: extensions
+
+        :type: :class:`collections.OrderedDict`
```

Comment:
Please don't subclass ordereddict, there's like a 0% chance it consistenyl
does what you want. Also it doesn't meet the idea of hiding implementation
details. Finally, it exposes the data to let people mutate it, which we
want to avoid.

On Sat, Mar 21, 2015 at 8:52 PM, Donald Stufft <notifications@github.com>
wrote:

> In docs/x509.rst
> <https://github.com/pyca/cryptography/pull/1769#discussion_r26898655>:
>
> > @@ -196,6 +196,24 @@ X.509 Certificate Object
> >              >>> isinstance(cert.signature_hash_algorithm, hashes.SHA256)
> >              True
> >
> > +    .. attribute:: extensions
> > +
> > +        :type: :class:`collections.OrderedDict`
>
> That class could just be a subclass of OrderedDict I think?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1769/files#r26898655>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084